File: pdlearning/src/microservices/course-microservice/Api/Application/Models/SendAnnouncementEmailTemplateModel.cs
namespace Microservice.Course.Application.Models
{
    public class SendAnnouncementEmailTemplateModel : EmailTemplateModel
    {
        public string UserNameTagValue
        {
            get => Tags.ContainsKey(nameof(UserNameTagValue)) ? Tags[nameof(UserNameTagValue)] : string.Empty;

            set => Tags[nameof(UserNameTagValue)] = value;
        }

        public string CourseTitleTagValue
        {
            get => Tags.ContainsKey(nameof(CourseTitleTagValue)) ? Tags[nameof(CourseTitleTagValue)] : string.Empty;

            set => Tags[nameof(CourseTitleTagValue)] = value;
        }

        public string CourseCodeTagValue
        {
            get => Tags.ContainsKey(nameof(CourseCodeTagValue)) ? Tags[nameof(CourseCodeTagValue)] : string.Empty;

            set => Tags[nameof(CourseCodeTagValue)] = value;
        }

        public string CourseAdminNameTagValue
        {
            get => Tags.ContainsKey(nameof(CourseAdminNameTagValue)) ? Tags[nameof(CourseAdminNameTagValue)] : string.Empty;

            set => Tags[nameof(CourseAdminNameTagValue)] = value;
        }

        public string CourseAdminEmailTagValue
        {
            get => Tags.ContainsKey(nameof(CourseAdminEmailTagValue)) ? Tags[nameof(CourseAdminEmailTagValue)] : string.Empty;

            set => Tags[nameof(CourseAdminEmailTagValue)] = value;
        }

        public string ListSessionTagValue
        {
            get => Tags.ContainsKey(nameof(ListSessionTagValue)) ? Tags[nameof(ListSessionTagValue)] : string.Empty;

            set => Tags[nameof(ListSessionTagValue)] = value;
        }

        public string DetailUrlTagValue
        {
            get => Tags.ContainsKey(nameof(DetailUrlTagValue)) ? Tags[nameof(DetailUrlTagValue)] : string.Empty;

            set => Tags[nameof(DetailUrlTagValue)] = value;
        }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/models/send-announcement-email-template.model.ts
import { EmailTemplateModel, IEmailTemplateModel } from './email-template.model';

export interface ISendAnnouncemmentEmailTemplateModel extends IEmailTemplateModel {
  userNameTagValue?: string;
  courseTitleTagValue?: string;
  courseCodeTagValue?: string;
  courseAdminNameTagValue?: string;
  courseAdminEmailTagValue?: string;
  listSessionTagValue?: string;
  detailUrlTagValue?: string;
}

export class SendAnnouncemmentEmailTemplateModel extends EmailTemplateModel implements ISendAnnouncemmentEmailTemplateModel {
  public userNameTagValue?: string;
  public courseTitleTagValue?: string;
  public courseCodeTagValue?: string;
  public courseAdminNameTagValue?: string;
  public courseAdminEmailTagValue?: string;
  public listSessionTagValue?: string;
  public detailUrlTagValue?: string;

  constructor(data?: ISendAnnouncemmentEmailTemplateModel) {
    super(data);

    if (data == null) {
      return;
    }
    this.userNameTagValue = data.userNameTagValue;
    this.courseTitleTagValue = data.courseTitleTagValue;
    this.courseCodeTagValue = data.courseCodeTagValue;
    this.courseAdminNameTagValue = data.courseAdminNameTagValue;
    this.courseAdminEmailTagValue = data.courseAdminEmailTagValue;
    this.listSessionTagValue = data.listSessionTagValue;
    this.detailUrlTagValue = data.detailUrlTagValue;
  }
}
